Model rework to support many-to-one

IReallyHateLogins opened the pull request:
Added lombok to remove many and many default setters-getters and such
Added repository for mailboxes
Moved folders to mailbox repository
Folders now stored as string and parsed at runtime, will do for now
Added MailFolder to parse folders at runtime, will work on correct folder representation later
Commented out mail functions, will move them to mailbox and filder later as aprt of my task
Added endpoints for mailbox, for testing

IReallyHateLogins committed:
Model rework to support many-to-one

Added lombok to remove many and many default setters-getters and such
Added repository for mailboxes
Moved folders to mailbox repository
Folders now stored as string and parsed at runtime, will do for now
Added MailFolder to parse folders at runtime, will work on correct folder representation later
Commented out mail functions, will move them to mailbox and filder later as aprt of my task
Added endpoints for mailbox, for testing

snowzhop reviewed (approved):
Well, I like when we delete Java code. 
Approved!

Egridnev committed:
Database config added

Egridnev committed:
Databases profiles added

Added ability to choose dev and test db config
Added different properties files for test and dev configurations
Added dependencie for mysql connector
Now we can choose which database to use, but you still need to install MySql locally

Egridnev committed:
Removed unnecessary dependecie

IReallyHateLogins committed:
Merge pull request #28 from Valbod-I-am-a-Swiftie/database-connection

Database connection

Egridnev reviewed (approved):
Well done!!!
